### PR TITLE
Gutenberg: Fix displaying validation warning and usage of PHP function

### PR DESCRIPTION
**Pull request by miina**
Makes use of `core/notices`. Updates creating a warning notice.

See #1604.

### CODE REVIEW

**Comment by kienstra** [comment]
**Sorry, Will Review Tomorrow**

Hi @miina,
Sorry for the delay. Would it be alright if I reviewed this tomorrow?

**Comment by miina** [comment]
@kienstra Of course!

**Comment by kienstra** [comment]
**Sorry For The Delay**

Hi @miina,
Thanks a lot for fixing this, and sorry for the delay in reviewing it.

**Comment by kienstra** [comment]
**Looks Good, Request For Review**

Hi @westonruter,
Could you please review this when you have a chance?

@miina's PR looks good, and it fixes the issue, as I understand it.

It would be great to have this PR in `1.0`, but when I changed the target branch from `develop` to `1.0`, it looks like this might have been branched off another branch, as more commits appeared.

Here's a [feature branch](https://github.com/Automattic/amp-wp/compare/1.0...fix/cherry-picked-displaying-validation-warning?expand=1) with Miina's commits cherry-picked onto a branch off of `1.0`.

# Before
before-console-errors

# After
after-gutenberg-expected

**Comment by westonruter** [comment]
I'm re-basing the commits here onto the `1.0` branch.

File: assets/js/amp-block-validation.js [hunk 1]
```diff
@@ -249,16 +249,20 @@ var ampBlockValidation = ( function() { // eslint-disable-line no-unused-vars
...
+				noticeOptions.actions = [
+					{
+						label: wp.i18n.__( 'Review issues', 'amp' ),
+						url: ampValidity.review_link
+					}
+				];
+			}
 
-			module.validationWarningNoticeId = wp.data.dispatch( 'core/editor' ).createWarningNotice( noticeElement, { spokenMessage: noticeMessage } ).notice.id;
+			wp.data.dispatch( 'core/notices' ).createNotice( 'warning', noticeMessage, noticeOptions );
```
Comment (westonruter) [inline assets/js/amp-block-validation.js]: I'm getting an error here now when attempting to edit an existing post:

image

Comment (miina) [reply assets/js/amp-block-validation.js]: @westonruter Weird, that would indicate as if `core/notices` wouldn't exist. Is this with the latest version of Gutenberg?

Comment (westonruter) [reply assets/js/amp-block-validation.js]: Humm. I'm not seeing the issue anymore. I opened an incognito window and everything. I'll report if I see it again.

**Pull request merged**